pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -17,31 +17,35 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"context"
 	"fmt"
 	"time"
 
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
 	"k8s.io/klog"
 	"k8s.io/kubernetes/pkg/scheduler/apis/config"
 	"k8s.io/kubernetes/pkg/scheduler/internal/cache"
+	schedulerutil "k8s.io/kubernetes/pkg/scheduler/util"
 )
 
 // framework is the component responsible for initializing and running scheduler
 // plugins.
 type framework struct {
-	registry         Registry
-	nodeInfoSnapshot *cache.NodeInfoSnapshot
-	waitingPods      *waitingPodsMap
-	plugins          map[string]Plugin // a map of initialized plugins. Plugin name:plugin instance.
-	queueSortPlugins []QueueSortPlugin
-	prefilterPlugins []PrefilterPlugin
-	reservePlugins   []ReservePlugin
-	prebindPlugins   []PrebindPlugin
-	postbindPlugins  []PostbindPlugin
-	unreservePlugins []UnreservePlugin
-	permitPlugins    []PermitPlugin
+	registry              Registry
+	nodeInfoSnapshot      *cache.NodeInfoSnapshot
+	waitingPods           *waitingPodsMap
+	pluginNameToWeightMap map[string]int32
```

Comment:
Let's make this `map[string]int`. This way we pay the cast overhead once, as opposed to casting every time when we run score plugins.

Reply:
Done.